api/base/permissions.py

```diff
@@ -53,6 +54,12 @@ def _get_scopes(self, request, view):
             except AttributeError:
                 raise ImproperlyConfigured('TokenHasScope requires the view to define the '
                                            'required_read_scopes attribute')
+            assert is_iterable_but_not_string(view.required_read_scopes), \
+                'The required_read_scopes must be an iterable of CoreScopes'
+            if view.required_read_scopes and isinstance(view.required_read_scopes[0], tuple):
```

Comment:
Yeah, but this change would have been much better three rounds of code review ago. Since it took so long, and since it's such a small amount of optimization, I do not recommend trying to move this code. Since it will be a while to  this merged in, replacing it with a try-except would be fine.